openedx_tagging/core/tagging/system_defined_taxonomies/object_tags.py

```diff
@@ -0,0 +1,200 @@
+from typing import List
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.core.exceptions import FieldDoesNotExist
+from django.db import models
+
+
+from openedx_tagging.core.tagging.models import (
+    Tag,
+    Taxonomy,
+    OpenObjectTag,
+    ClosedObjectTag,
+)
+from openedx_tagging.core.tagging.registry import register_object_tag_class
+
+
+class SystemDefinedObjectTagMixin:
+    """
+    Mixing for ObjectTags used on all system defined taxonomies
+
+    `system_defined_taxonomy_name``is used to connect the 
+    ObjectTag with the system defined taxonomy.
+    This is because there can be several ObjectTags
+    for the same Taxonomy, ex:
+    
+    - LanguageCourseObjectTag
+    - LanguageBlockObjectTag
+
+    On the example, there are ObjectTags for the same Language
+    taxonomy but with different objects.
```

Comment:
@pomegranited I spoted this case, there can be several `ObjectTags` for the same Taxonomy. This makes `_object_tag_class` not work in this cases. I've solved that here: https://github.com/open-craft/openedx-learning/pull/3/commits/b67ca9a6061fce7fe7f1d61f51e06bfb757edf25

What do you think?

Reply:
@ChrisChV I'm confused.. can't we create separate Taxonomy instances for each of the System taxonomies, and set a different Taxonomy._object_tag_class for each of them? 

I went back and forth on whether we even needed the ObjectTag registry, because in theory, every new Taxonomy could just specify its ideal ObjectTagClass on creation. But I thought that might add difficulty for the Taxonomy Admins, because they'd have to understand the ObjectTag classes. But since we're doing the System taxonomies as separate hardcoded things, we can include the `_object_tag_class` in the fixtures.

Reply:
Ok @ChrisChV , I think I've addressed this issue with the latest changes.. see https://github.com/openedx/edx-platform/pull/32661#discussion_r1260655868.